test/playlist-loader.test.js

```diff
@@ -691,7 +691,7 @@ QUnit.test('aborts in-flight playlist refreshes when switching', function(assert
   assert.strictEqual(this.requests[0].aborted, true, 'aborted refresh request');
   assert.ok(!this.requests[0].onreadystatechange,
            'onreadystatechange handlers should be removed on abort');
-  assert.strictEqual(loader.state, 'SWITCHING_MEDIA', 'updated the state');
+  assert.strictEqual(loader.state, 'HAVE_METADATA', 'the state is set accoring to the startingState');
```

Comment:
Because when the refresh request is aborted, ```playlistRequestError``` is called and we reset the loader.state to startingState(HAVE_METADATA). so that no longer is set to SWITCHING_MEDIA. I think if switching happens without error, the loader.state as SWITCHING_MEDIA won't be affected.